src/main/java/com/nike/cerberus/domain/environment/EnvironmentData.java

```diff
@@ -0,0 +1,160 @@
+
+    public String getEnvironmentName() {
+        return environmentName;
+    }
+
+    public void setEnvironmentName(String environmentName) {
+        this.environmentName = environmentName;
+    }
+
+    public String getAdminIamRoleArn() {
+        return adminIamRoleArn;
+    }
+
+    public void setAdminIamRoleArn(String adminIamRoleArn) {
+        this.adminIamRoleArn = adminIamRoleArn;
+    }
+
+    public String getRootIamRoleArn() {
+        return rootIamRoleArn;
+    }
+
+    public void setRootIamRoleArn(String rootIamRoleArn) {
+        this.rootIamRoleArn = rootIamRoleArn;
+    }
+
+    public String getCmsIamRoleArn() {
+        return cmsIamRoleArn;
+    }
+
+    public void setCmsIamRoleArn(String cmsIamRoleArn) {
+        this.cmsIamRoleArn = cmsIamRoleArn;
+    }
+
+    public String getDatabasePassword() {
+        return databasePassword;
+    }
+
+    public void setDatabasePassword(String databasePassword) {
+        this.databasePassword = databasePassword;
+    }
+
+    public LinkedList<CertificateInformation> getCertificateData() {
+        return certificateInfoList;
+    }
+
+    public void addNewCertificateData(CertificateInformation certificateData) {
+        getCertificateData().add(certificateData);
+    }
+
+    public void removeCertificateInformationByName(String identityManagementCertificateName) {
+        List<CertificateInformation> certsInfoToBeDeleted = certificateInfoList.stream()
+                .filter(certificateInformation ->
+                        certificateInformation.getCertificateName().equals(identityManagementCertificateName))
+                .collect(Collectors.toList());
+
+        certificateInfoList.removeAll(certsInfoToBeDeleted);
+    }
+
+    public Map<Regions, RegionData> getRegionData() {
+        return regionData;
+    }
+
+    public void addRegionData(Regions region, RegionData rData) {
+        regionData.put(region, rData);
+    }
+
+    /**
+     * @return List of all the regions that are configured for storing config.
+     */
+    @JsonIgnore
+    public List<Regions> getConfigRegions() {
+        List<Regions> configRegions = new LinkedList<>();
+        regionData.forEach((region, rData) -> {
+            if (rData.getEnvironmentDataSecureDataKmsCmkArn().isPresent()
+                    && rData.getConfigBucket().isPresent()) {
+                configRegions.add(region);
+            }
+        });
+        return configRegions;
+    }
+
+    @JsonIgnore
+    public Regions getPrimaryRegion() {
+        final Regions[] primaryRegion = {null};
+        regionData.forEach((region, regionData) -> {
+            if (regionData.isPrimary()) {
+                primaryRegion[0] = region;
+            }
+        });
+        if (primaryRegion[0] == null) {
+        throw new RuntimeException("Failed to find primary region in region specific state");
+    }
+        return primaryRegion[0];
+    }
+
+    public void addCmksCmkForRegion(Regions region, String arn) {
```

Comment:
"CmksCmk" ?

Also, "arn" could be "cmkArn" or have Javadoc to be more obvious what ARN we're dealing with.